collect_app/src/main/java/org/odk/collect/android/formmanagement/ServerFormDownloader.java

```diff
@@ -102,17 +98,18 @@ private void processOneForm(ServerFormDetails fd, FormDownloaderListener stateLi
 
             // download media files if there are any
             if (fd.getManifest() != null && !fd.getManifest().getMediaFiles().isEmpty()) {
-                downloadMediaFiles(tempMediaPath, stateListener, fd.getManifest().getMediaFiles(), tempDir, fileResult.file.getName());
+                FormMediaDownloader mediaDownloader = new FormMediaDownloader(formsRepository, formSource);
```

Comment:
I think you can inject `FormMediaDownloader` via constructor instead of creating it here.

Reply:
Right now I'd opt not to as `FormMediaDownloder` doesn't have any unit tests (it's behaviour is driven out/verified by `ServerFormDownloader`) and the component using it doesn't have tests that swap it out for something else. In short, it's a "detail" of `ServerFormDownloader` still. It might be that we want to get to that point soon (`ServerFormDownloader` taking in "smaller" use case objects and just testing it does the right thing with them but didn't feel like that was a path I wanted to go down right now - we can probably shift tests over if we feel like it next time we touch media downloading. For the moment I just wanted to move the code around to remove noise from the other file. 

One thought I have is we might want to actually break the whole thing into a pipeline of different use cases, so it could be that `FormMediaDownloader` ends up as a step in that.